app/components/ScrollToTop.ts

```diff
@@ -25,8 +27,9 @@ export default function ScrollToTop({ children }: Props) {
     ) {
       return;
     }
-    window.scrollTo(0, 0);
+    (scrollContainerRef?.current || window).scrollTo(0, 0);
```

Comment:
The problem was this hardcoded to resetting `window` scroll, but on mobile we're in a wrapper.